### ISSUE TITLE
GRPO zero3 训练qwen3.5 9b rollout一直为乱码或者全部是!!!!!!!!!!!!!!，导致无法训练

### ISSUE THREAD

**Issue by UCC-team** [modelscope/ms-swift#9306]
### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new bug report. / 我已经搜索过现有的 issues，确认这是一个新的 bug report。

### Bug Description / Bug 描述

现象与[7112]( https://github.com/modelscope/ms-swift/issues/7112) 一致。

使用--deepspeed zero3输出乱码或者输出全部"!!!!!!!!!!!!!!"，但是切换成--deepspeed zero1就完全正常。
但是因为显存限制，我必须使用zero3

### How to Reproduce / 如何复现

swift rlhf \
    --rlhf_type grpo \
    --model ${model_path} \
    --dataset ${dataset} \
    --tuner_type full \
    --per_device_train_batch_size 1 \
    --gradient_accumulation_steps 4 \
    --num_generations 4 \
    --max_grad_norm 0.5 \
    --attn_impl flash_attn \
    --torch_dtype bfloat16 \
    --temperature 1.0 \
    --external_plugins plugin.py \
    --reward_funcs soft_overlong repetition tool_reward \
    --max_completion_length 2048 \
    --soft_cache_length 768 \
    --repetition_n_grams 5 \
    --num_train_epochs 8 \
    --epsilon_high 0.28 \
    --loss_type 'grpo' \
    --learning_rate 1e-6 \
    --lr_scheduler_type 'constant' \
    --save_strategy steps \
    --save_steps 16 \
    --save_total_limit 5 \
    --save_only_model false \
    --data_seed 12 \
    --sleep_level 0 \
    --log_completions true \
    --log_entropy true \
    --top_entropy_quantile 0.2 \
    --delta 4 \
    --logging_steps 1 \
    --overlong_filter true \
    --dynamic_sample true \
    --num_iterations 1 \
    --output_dir ${checkpoint_output} \
    --split_dataset_ratio 0 \
    --offload_optimizer false \
    --offload_model false \
    --use_vllm true \
    --vllm_mode colocate \
    --vllm_gpu_memory_utilization 0.4 \
    --vllm_enable_prefix_caching False \
    --vllm_max_model_len 32000 \
    --vllm_tensor_parallel_size 8 \
    --vllm_use_async_engine false \
    --async_generate false \
    --beta 0.0 \
    --deepspeed zero3

**Comment by hjh0119** [comment]
plz provide your hardware and environment infos

**Comment by UCC-team** [comment]
> plz provide your hardware and environment infos

A100 80G
ms_swift版本：
Version: 4.2.0.dev0

**Comment by hjh0119** [comment]
Please provide more environment information, and check whether the CUDA_LAUNCH_BLOCKING environment variable is set. If so, please unset it.

**Comment by UCC-team** [comment]
> Please provide more environment information, and check whether the CUDA_LAUNCH_BLOCKING environment variable is set. If so, please unset it.

CUDA_LAUNCH_BLOCKING没有设置，唯一设置的环境变量是：
export PYTORCH_CUDA_ALLOC_CONF="expandable_segments:True"

（上下文比较上大概 32000）

**Comment by UCC-team** [comment]
在PPU和A800上也可以稳定复现。具体表现为：

deepspeed zero3时，PPU上输出为乱码，A100和A800上从第二个step才开始输出为全部!!!!!!
PPU：

<img width="1799" height="193" alt="Image" src="https://github.com/user-attachments/assets/268ee486-01c0-4128-b11c-052a0aa9f8cd" />

A800：

<img width="1804" height="138" alt="Image" src="https://github.com/user-attachments/assets/f6f90848-61c4-477f-a9a0-aa934613e945" />

**Comment by Jingwei-Bao** [comment]
Same Error. 在运行OPSD训练时遇到问题：
NPROC_PER_NODE=7 \
CUDA_VISIBLE_DEVICES=1,2,3,4,5,6,7 \
swift rlhf \
    --rlhf_type gkd \
    --model Qwen3.5-4B \
    --teacher_model Qwen3.5-4B \
    --gkd_logits_topk 64 \
    --tuner_type full \
    --target_modules all-linear \
    --use_vllm true \
    --vllm_mode colocate \
    --vllm_gpu_memory_utilization 0.4 \
    --sleep_level 1 \
    --dataset 'opsd_dataset.jsonl' \
    --split_dataset_ratio 0.05 \
    --lmbda 1.0 \
    --beta 1.0 \
    --temperature 1.0 \
    --sft_alpha 0 \
    --torch_dtype bfloat16 \
    --num_train_epochs 4 \
    --per_device_train_batch_size 1 \
    --gradient_accumulation_steps 1 \
    --learning_rate 2e-5 \
    --warmup_ratio 0.05 \
    --save_steps 100 \
    --eval_steps 100 \
    --save_total_limit 10 \
    --logging_steps 1 \
    --max_length 8000 \
    --max_completion_length 8000 \
    --save_only_model true \
    --gradient_checkpointing true \
    --deepspeed zero3 \
    --attn_impl flash_attention_2 \
    --report_to wandb \
    --output_dir yiwu_llm/output/Qwen3.5-4B/exp_sft_opsd
拿ckpt推理时，发现：

<img width="1405" height="526" alt="Image" src="https://github.com/user-attachments/assets/6980e396-bbe2-4e8d-b5f3-f457e58813a4" />

**Comment by Jingwei-Bao** [comment]
L20 48G
ms_swift版本：
Version: 4.0.2

**Comment by Hao3zz** [comment]
same error occurs. The first step was normal, after the loss output, and then completions all became "!!!!!!!!" till reach the length limit. I am using vllm==0.17.0, 
 transformers==5.2.0, ms-swift==4.2.0.dev0

export LOG_LEVEL=DEBUG
export PYTORCH_CUDA_ALLOC_CONF='expandable_segments:True'
CUDA_VISIBLE_DEVICES=0,1,2,3 \
NPROC_PER_NODE=4 \
swift rlhf \
    --rlhf_type grpo \
    --model /dockerdata/Qwen3.5-9B \
    --external_plugins examples/train/grpo/plugin/plugin.py \
    --reward_funcs external_stageone_grpo \
    --enable_thinking false \
    --split_dataset_ratio 0.1 \
    --use_vllm true \
    --move_model_batches 10 \
    --vllm_mode colocate \
    --vllm_gpu_memory_utilization 0.4 \
    --vllm_tensor_parallel_size 4 \
    --vllm_max_model_len 2048 \
    --sleep_level 1 \
    --tuner_type full \
    --torch_dtype bfloat16 \
    --dataset '/group/40179/leehohli/data/train_data/grpo_train_easy_200.json' \
    --load_from_cache_file true \
    --max_length 2048 \
    --max_completion_length 256 \
    --num_train_epochs 1 \
    --per_device_train_batch_size 2 \
    --per_device_eval_batch_size 2 \
    --gradient_accumulation_steps 4 \
    --learning_rate 5e-6 \
    --lr_scheduler_type cosine \
    --save_steps 20 \
    --eval_steps 20 \
    --save_total_limit 100 \
    --logging_steps 1 \
    --warmup_ratio 0.1 \
    --dataloader_num_workers 4 \
    --num_generations 8 \
    --temperature 1.0 \
    --deepspeed zero3_offload \
    --log_completions true \
    --max_grad_norm 1.0 \
    --epsilon 0.2 \
    --epsilon_high 0.28 \
    --scale_rewards none

**Comment by hjh0119** [comment]
run normally on my side 

<img width="1524" height="882" alt="Image" src="https://github.com/user-attachments/assets/c97a87a8-df47-4c3c-8531-d8876e08ba8d" />


```
SYSTEM_PROMPT="""You are a helpful math assistant. Solve the problem step by step and put your final answer within \\boxed{}."""


LOG_LEVEL=DEBUG \
PYTORCH_CUDA_ALLOC_CONF='expandable_segments:True' \
CUDA_VISIBLE_DEVICES=4,5,6,7 \
NPROC_PER_NODE=4 \
swift rlhf \
    --rlhf_type grpo \
    --model Qwen/Qwen3.5-9B \
    --reward_funcs accuracy \
    --enable_thinking false \
    --use_vllm true \
    --move_model_batches 10 \
    --vllm_mode colocate \
    --vllm_gpu_memory_utilization 0.4 \
    --vllm_tensor_parallel_size 4 \
    --vllm_max_model_len 8192 \
    --sleep_level 1 \
    --tuner_type full \
    --torch_dtype bfloat16 \
    --dataset open-r1/DAPO-Math-17k-Processed#100 \
    --system "$SYSTEM_PROMPT" \
    --load_from_cache_file true \
    --max_length 4096 \
    --max_completion_length 4096 \
    --num_train_epochs 1 \
    --per_device_train_batch_size 2 \
    --gradient_accumulation_steps 4 \
    --learning_rate 5e-6 \
    --lr_scheduler_type cosine \
    --save_steps 20 \
    --eval_steps 20 \
    --save_total_limit 100 \
    --logging_steps 1 \
    --warmup_ratio 0.1 \
    --dataloader_num_workers 4 \
    --num_generations 8 \
    --temperature 1.0 \
    --deepspeed zero3_offload \
    --log_completions true \
    --max_grad_norm 1.0 \
    --report_to swanlab \
    --epsilon 0.2 \
    --epsilon_high 0.28 \
    --scale_rewards none
```

**Comment by Hao3zz** [comment]
it's kinda werid, I use zero1 instead of zero3_offload and it works well as author mentioned, but i had to use lora due to the gpu limitation

**Comment by Asunatan** [comment]
这个问题是多模态GRPO的问题，我测试纯文本GRPO是正常的

**Comment by hjh0119** [comment]
please retry with latest swift, feel free to reopen or create a new issue if it the problem persists.
